components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -86,6 +86,16 @@ private void initializeDataPublisher() {
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
 
         String channelId = ctx.channel().id().asLongText();
+
+        // This block is for the health check of the ports 8099 and 9099
+        if (msg instanceof FullHttpRequest && (!((FullHttpRequest) msg).headers().contains(HttpHeaders.UPGRADE))) {
+            FullHttpResponse httpResponse = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+            httpResponse.headers().set("content-type", "text/plain; charset=UTF-8");
+            httpResponse.headers().set("content-length", httpResponse.content().readableBytes());
+            ctx.writeAndFlush(httpResponse);
+            return;
+        }
```

Comment:
Does this have an integration test? If not shall we add one to the same Websocket test cases in product-apim? @binodmx @Arshardh

Reply:
There is no IT for this.

However, it's better to add separate test case for all the health checks in [1] including these newly introduced web socket ports health checks. WDYT?

[1] https://apim.docs.wso2.com/en/latest/install-and-setup/setup/deployment-best-practices/basic-health-checks/#api-m-runtime-health-checks

Reply:
+1 thats better

Reply:
Test case [1] created for health check APIs including web socket ports.

[1] https://github.com/wso2/product-apim/pull/12227